relayer/spy_relayer/src/helpers/redisHelper.ts

```diff
@@ -428,3 +385,51 @@ export async function monitorRedis(metrics: PromHelper) {
     await sleep(TEN_SECONDS);
   }
 }
+
+/** Check to see if a queue is in the listener memory queue before redis */
+export async function checkQueue(key: string): Promise<string | null> {
+  try {
+    const backupQueue = getBackupQueue();
+    const queuedRecord = backupQueue.find((record) => {
+      record[0] === key;
+    });
+
+    if (queuedRecord) {
+      logger.debug("VAA was already in the listener queue");
+      return "VAA was already in the listener queue";
+    }
+
+    const rClient = await connectToRedis();
+    if (!rClient) {
+      logger.error("Failed to connect to redis");
+      return null;
+    }
+    /**
+     * TODO: Pretty sure this code never ever worked for checking if a key is in redis.
+     *
+     * The `key` variable is `chainId + ":" + address`, but the actual redis keys come
+     * from serializing the StoreKey type into a stringified json representation.
+     */
```

Comment:
I'd love to fix this and the easiest way (though my least fav) is making the memory queue use the same json representation as the json set as the redis key, I just don't love that.

More preferable to me would be setting this format however, it would mess up anything outstanding in redis when the upgrade happens:

    {chainId}/{emitterAddress}/{sequenceNumber}

Reply:
Could we use `const storeKeyJson = storeKeyToJson(storeKeyFromParsedVAA(parsedVaa));` as the `key` - making the memory queue use the same key format as the redis store. I don't see why it'd be so bad

Reply:
I can totally do that, just think the other way is easier to read. But you've made a totally legit suggestion, so I'm going to go ahead and do it. Thanks!